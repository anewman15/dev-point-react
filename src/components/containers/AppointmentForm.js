/* eslint-disable jsx-a11y/no-autofocus */
import { PropTypes } from 'prop-types';
import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import createAppointment from '../../sandbox/createAppointment';

function AppointmentForm({ devId, devConfLink }) {
  const initAppointmentInfo = {
    appointment_guest_id: devId,
    location: '',
    time: '',
    video_conf_link: devConfLink,
  };

  const [appoinmentInfo, setAppointmentInfo] = useState(initAppointmentInfo);
  const history = useHistory();

  const handleChange = e => {
    setAppointmentInfo({
      ...appoinmentInfo,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = e => {
    e.preventDefault();
    createAppointment(appoinmentInfo)
      .then(response => response.json())
      .then(data => {
        if (data.status === 'created') {
          setAppointmentInfo(initAppointmentInfo);
          history.push('/appointments');
        }
      });
  };

  return (
    <div className="container my-6 mx-auto p-6 bg-warning-500 bg-opacity-60 rounded">
      <h2 className="text-l md:text-xl text-primary-800 text-center font-semibold">Book an Appointment</h2>

      <hr className="my-4 border-1 border-primary-600" />

      <div className="flex flex-wrap justify-start items-center">
        <form className="" onSubmit={handleSubmit}>
          <div className="field">
            <label className="label" htmlFor="location">
              Location
              <div className="control">
                <input
                  type="text"
                  className="input"
                  name="location"
                  onChange={handleChange}
                  placeholder="A place, or just Online"
                  required
                  autoFocus
                />
              </div>
            </label>
          </div>
          <div className="field">
            <label className="label" htmlFor="time">
              Time
              <div className="control">
                <input
                  type="datetime-local"
                  className="input"
                  name="time"
                  onChange={handleChange}
                  required
                />
              </div>
            </label>
          </div>
          <div className="control">
            <button
              type="submit"
              className="btn btn-primary"
            >
              Book Now
            </button>
          </div>
        </form>
      </div>
    </div>
  );
}

AppointmentForm.propTypes = {
  devId: PropTypes.integer,
  userId: PropTypes.integer,
}.isRequired;

export default AppointmentForm;
