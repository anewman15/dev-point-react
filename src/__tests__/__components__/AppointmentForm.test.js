import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import AppointmentForm from '../../components/containers/AppointmentForm';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('AppointmentsForm', () => {
  it('renders the AppointmentForm component to the dom', () => {
    const tree = renderer
      .create(<AppointmentForm />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
