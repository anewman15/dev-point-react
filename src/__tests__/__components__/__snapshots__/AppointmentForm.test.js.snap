// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentsForm renders the AppointmentForm component to the dom 1`] = `
<form
  className="mx-6 mt-4 pb-6"
  onSubmit={[Function]}
>
  <div
    className="columns is-centered"
  >
    <div
      className="column is-four-fifths"
    >
      <div
        className="mx-6 field is-horizontal"
      >
        <div
          className="field-body is-grouped"
        >
          <div
            className="field"
          >
            <label
              className="label has-text-centered has-text-white"
              htmlFor="location"
            >
              Location
              <div
                className="control"
              >
                <input
                  className="input is-primary is-rounded has-text-success"
                  name="location"
                  onChange={[Function]}
                  required={true}
                  type="text"
                />
              </div>
            </label>
          </div>
          <div
            className="field is-horizontal"
          >
            <label
              className="label has-text-centered has-text-white"
              htmlFor="time"
            >
              Time
              <div
                className="control"
              >
                <input
                  className="input is-primary is-rounded has-text-success"
                  name="time"
                  onChange={[Function]}
                  required={true}
                  type="datetime-local"
                />
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column is-vcentered"
    >
      <div
        className="mx-6 field is-horizontal is-vcentered"
      >
        <div
          className="control"
        >
          <button
            className="button is-white has-text-success is-rounded is-uppercase is-fullwidth mt-4 p-5 has-text-weight-bold"
            type="submit"
          >
            Book Now
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
